bokeh/application/handlers/code_runner.py

```diff
@@ -181,7 +181,7 @@ def new_module(self) -> ModuleType | None:
 
         module_name = 'bokeh_app_' + make_globally_unique_id().replace('-', '')
         module = ModuleType(module_name)
-        module.__dict__['__file__'] = os.path.abspath(self._path)
+        module.__dict__['__file__'] = os.path.abspath(os.path.expanduser(self._path))
```

Comment:
AFAIK this value is only used for diagnostic output, and I also believe that the only way the `self._path` value originally gets set is from values supplied at the command line, which will already be shell expanded (so this is not necessary).